src/NAME/Elasticsearch/ElasticsearchVersionResolver.cs

```diff
@@ -0,0 +1,65 @@
+using NAME.Core;
+using NAME.Core.Exceptions;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Net;
+using System.Threading.Tasks;
+
+namespace NAME.Elasticsearch
+{
+    internal class ElasticsearchVersionResolver : ConnectedVersionResolver
+    {
+        private IConnectionStringProvider connectionStringProvider;
+
+        public ElasticsearchVersionResolver(IConnectionStringProvider connectionStringProvider, int connectTimeout, int readWriteTimeout)
+            : base(connectTimeout, readWriteTimeout)
+        {
+            this.connectionStringProvider = connectionStringProvider;
+        }
+
+        public override async Task<IEnumerable<DependencyVersion>> GetVersions()
+        {
+            var versions = new List<DependencyVersion>();
+            var connectionString = string.Empty;
+
+            if (!this.connectionStringProvider.TryGetConnectionString(out connectionString))
+            {
+                throw new ConnectionStringNotFoundException(this.connectionStringProvider.ToString());
+            }
+
+            HttpWebResponse response;
+            try
+            {
+                var request = this.GetHttpWebRequest(connectionString, SupportedDependencies.Elasticsearch.ToString());
```

Comment:
Although you set the `ContinueTimeout` in the `GetHttpWebRequest`, it is not enough, because the Elasticsearch HTTP server may accept the connection and not send the response, so we need a read write timeout here, take a look at the [ServiceVersionResolver](https://github.com/nosinovacao/name-sdk/blob/cd56d886a15d683ae55a3ab472ec17ee1d040142/src/NAME/Service/ServiceVersionResolver.cs#L161) for clues on how to do that.